sklearn/preprocessing/data.py

```diff
@@ -1956,3 +1967,222 @@ def transform(self, X):
         """
         return _transform_selected(X, self._transform,
                                    self.categorical_features, copy=True)
+
+
+class CountFeaturizer(BaseEstimator, TransformerMixin):
+
+    """
+    Adds in a new feature column containing the number of occurences of
```

Comment:
I probably need to read any references (which should be listed in the docstring below), but I don't really get why this technique works. I have seen similar where a feature is added for each class (i.e. each value of `y`) counting the frequency of features given each class.

Without being conditional on `y`, your transformer seems to be bluntly measuring density, and so is suited only to problems where the classes can be differentiated by their density in particular features.